src/SourceControl-GitHub/APIWrapper.ts

```diff
@@ -0,0 +1,101 @@
+import { Octokit } from "octokit";
+import { Issue } from "./Issue";
+import { Milestone } from "./Milestone";
+import { Organization } from "./Organization";
+import { PullRequest } from "./PullRequest";
+import { Repository } from "./Repository";
+
+// authenticate via personal token for now
+const octokit = new Octokit({ auth: process.env._AUTHTOKEN });
+
+export class APIWrapper {
+  /**
+   * GetOrganization - queries GitHub for Organization information, resolving all
+   *                   sub info, such as Repositories, Issues, etc.
+   *
+   * @param  {type} orgName: string organization name
+   * @return {type}                 a GitHub organization
+   */
+  static async GetOrganization(orgName: string): Promise<Organization> {
+    const {
+      data: { id },
+    } = await octokit.rest.orgs.get({ org: orgName });
+
+    const repositories: { [key: string]: Repository } = {};
+
+    // if > 100 repos, will be paged
+    const respRepos = await octokit.rest.repos.listForOrg({ org: orgName });
+
+    for (let i = 0; i < respRepos.data.length; i++) {
+      let repoName = respRepos.data[i].name;
+      let repoId = respRepos.data[i].id.toString();
+      const repo = await this.GetRepository(orgName, repoName, repoId);
+      repositories[repoName] = repo;
+    }
+
+    return new Organization(orgName, id.toString(), repositories);
+  }
+
+  static async GetRepository(
+    orgName: string,
+    repoName: string,
+    repoId: string
+  ): Promise<Repository> {
+    const repo = new Repository(repoId, repoName, orgName);
+
+    // if > 100 issues, will be paged
+    const respIssues = await octokit.rest.issues.listForRepo({
+      owner: orgName,
+      repo: repoName,
+    });
+
+    respIssues.data
```

Comment:
Just wondering, could this be done in one iteration instead of 3?
Something like...
```js
 respIssues.data.forEach(i => {
  if (i.pull_request === null) {
    const issue = new Issue(i.id.toString(), i.title, i.state)
    repo.addTask(issue)
  }
})
```

Reply:
Certainly - good call!